src/FastExpressionCompiler/FastExpressionCompiler.cs

```diff
@@ -1432,8 +1439,10 @@ private static void EmitLoadParamArg(ILGenerator il, int paramIndex, bool asAddr
                 for (int i = 0, n = exprs.Count; i < n; i++)
                 {
                     var expr = exprs[i];
-                    if (!TryEmit(expr, expr.Type, paramExprs, il, ref closure, parent, i))
-                        return false;
+                    if (parent != ExpressionType.Block || 

```

Comment:
Could you combine both ifs and comment the condition, why it is here?

Reply:
Don't think the if's could be combined. the TryEmit should only be called when first if is not true

Reply:
Ok.